internal/hcp/internal/controllers/hcclink/status.go

```diff
@@ -0,0 +1,36 @@
+package hcclink
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+const (
+	StatusKey = "consul.io/hcc-link"
+
+	StatusLinked                      = "linked"
+	LinkedReason                      = "SUCCESS"
+	DisabledReasonResourceAPIsEnabled = "DISABLED"
```

Comment:
We'll be eventually also adding a configuration to disable the linking feature, so maybe we could change this to something specific to the V2 resources being enabled. Maybe `DISABLED_V2_RESOURCES_UNSUPPORTED` or something to that effect?

Reply:
good point!